doc/mimalloc-doc.h

```diff
@@ -730,6 +730,32 @@ bool mi_heap_check_owned(mi_heap_t* heap, const void* p);
 /// @see mi_heap_contains_block()
 /// @see mi_heap_get_default()
 bool mi_check_owned(const void* p);
+/// Type of deferred free functions.
+/// @param heap The heap.
+/// @param force If \a true all outstanding items should be freed.
+/// @param heartbeat A monotonically increasing count.
+/// @param arg Argument that was passed at registration to hold extra state.
+///
+/// @see mi_heap_register_deferred_free
+typedef void (mi_local_deferred_free_fun)(mi_heap_t* heap,bool force,unsigned long long heartbeat,void* arg);
+
+/// Register a deferred free function.
+/// @param heap The heap where deferred function should be registered.
+/// @param deferred_free Address of a deferred free-ing function or \a NULL to unregister.
+/// @param arg Argument that will be passed on to the deferred free function.
+///
+/// Some runtime systems use deferred free-ing, for example when using
```

Comment:
Can you specify the names of such runtime systems? It would be better for figuring out their characteristics.

Reply:
Comment for regular mi_register_deferred_free does not mention these systems. I guess I could mention JavaScriptCore that has lazy sweeping(aka deferred free) in GC and JikesRVM.

Reply:
That would be great. I wonder how GC implementations benefit from mimalloc derived infrastructure.

Reply:
> That would be great. I wonder how GC implementations benefit from mimalloc derived infrastructure.


Well I alreay use mimalloc in [starlight](https://github.com/starlight-js/starlight) and using mimalloc API like `mi_heap_check_owned` I managed to implement conservative root scanning, using `mi_heap_visit_blocks` I can easily walk the heap to sweep it or to serialize it. If this PR will get merged I could implement lazy sweeping or generational collection.